components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/impl/CommentsManagerImpl.java

```diff
@@ -17,10 +17,274 @@
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return comment;
+    }
+
+    @Override
+    public void deleteComment(int apAppCommentId) throws CommentManagementException {
+
+        Comment comment;
+        comment= getComment(apAppCommentId);
+
+        if (comment == null) {
+            throw new CommentManagementException(
+                        "Cannot delete a non-existing comment for the application with comment id" + apAppCommentId);
+        }
+        try {
+            ConnectionManagerUtil.beginDBTransaction();
+            ApplicationManagementDAOFactory.getCommentDAO().deleteComment(apAppCommentId);
+            ConnectionManagerUtil.commitDBTransaction();
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } catch (SQLException e) {
+            log.error("SQL Exception occurs.", e);
+        } catch (TransactionManagementException e) {
+            log.error("Transaction Management Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+    }
+
+    @Override
+    public Comment updateComment(Comment comment,int apAppCommentId) throws CommentManagementException, SQLException,
+            DBConnectionException {
+
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true);
+        validateComment(apAppCommentId,comment.getComment());
+
+        if (log.isDebugEnabled()) {
+            log.debug("Comment retrieval request is received for the comment id " +apAppCommentId);
+        }
+        comment.setModifiedAt(Timestamp.from(Instant.now()));
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            ApplicationManagementDAOFactory.getCommentDAO().getComment(apAppCommentId);
+            return ApplicationManagementDAOFactory.getCommentDAO().updateComment(apAppCommentId,
+                    comment.getComment(),comment.getModifiedBy(),comment.getModifiedAt());
+        } catch (SQLException e) {
+            log.error("SQL Exception occurs.", e);
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return ApplicationManagementDAOFactory.getCommentDAO().getComment(apAppCommentId);
+    }
+
+    public int getRatedUser(String uuid){
+
+        int ratedUsers=0;
+        if (log.isDebugEnabled()) {
+            log.debug("Get the rated users for the application release number"+uuid);
+        }
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            ratedUsers =ApplicationManagementDAOFactory.getCommentDAO().getRatedUser(uuid);
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } catch (ApplicationManagementDAOException e) {
+            log.error("Application Management Exception occurs.", e);
```

Comment:
Just logging the error is not sufficient, we need to notify upper layer. You may throw a wrapped exception of `CommentManagementException`. Fix other `catch` blocks in this class as well.